src/control/server/instance.go

```diff
@@ -44,6 +44,9 @@ import (
 	"github.com/daos-stack/daos/src/control/system"
 )
 
+type onStorageReadyFn func() error
+type onReadyFn func(ctx context.Context) error
```

Comment:
Hmm, no good reason. Those were added a bit organically and I was capturing the outer context in the closure, but accepting a context for both callback types makes sense. I think I still want to keep them as separate types in case they do diverge later though.